utils/__init__.py

```diff
@@ -16,5 +16,133 @@
 """
 
 from taskw_ng.warrior import TaskWarrior
+from .task import Task
+from enum import Enum
+from typing import Callable, Optional
 
-taskWarriorInstance = TaskWarrior()  # Create a TaskWarrior instance.
\ No newline at end of file
+def singleton(cls):
+    instances = {}
+    def getinstance(*args, **kwargs):
+        if cls not in instances:
+            instances[cls] = cls(*args, **kwargs)
+        return instances[cls]
+    return getinstance
+
+class SortMetric(Enum):
+    ID_ASCENDING           = 0
+    ID_DESCENDING          = 1
+    PRIORITY_ASCENDING     = 2
+    PRIORITY_DESCENDING    = 3
+    DESCRIPTION_ASCENDING  = 4
+    DESCRIPTION_DESCENDING = 5
+    # TODO: Add wayy more
+
+@singleton
+class TaskAPI:
+
+    @staticmethod
+    def _get_sort_params(metric: SortMetric) -> tuple[Callable[[Task], str | int], bool]:
+        """Meant to be called in `self._init_task_list()` by doing the following:
+        
+        ```
+        k, r = self._get_sort_params(self.sort_metric)
+        self.task_list.sort(key=k, reverse=r)
+        ```
+        That's it.
+
+        """
+        
+        def alpha_sorting(attr_name: str):
+            """Used whenever the attribute we are sorting by follows alphanumeric sorting rules.
+            
+            Ignores capitalization.
+
+            Examples: description, id, project.
+            Counterexamples: priority."""
+            
+            def subFn(t: Task):
+                return str(t[attr_name]).lower()
+            
+            return subFn
+
+        def priority_sorting(t: Task): 
+            match t.get_priority():
+                case 'H':
+                    return 1
+                case 'M':
+                    return 2
+                case 'L':
+                    return 3
+                case _:
+                    return 0
+
+        match metric:
+            case SortMetric.ID_ASCENDING:
+                return alpha_sorting('id'), False
```

Comment:
Do this format for all of the alphanumeric sorting and then remove the submethod. It'll be cleaner when we don't have a submethod with another submethod. Shoutout to 662 for making me realize this was a thing.

```suggestion
                return lambda t: str(t['id']).lower(), False
```

Reply:
I used a nested function because of code reuse stuff. We could just as well change the definition of alpha_sorting to use a lambda function, and id be down to do that.